### PR TITLE
[WFCORE-2262] Latest WildFly Elytron and Invocation Changes

### PR DESCRIPTION
**Pull request by darranl**
:bangbang: Commits from this branch are used in numerous other topic branches so please do not cherry-pick or rebase. :bangbang:

The following issues are resolved by this PR: -

https://issues.jboss.org/browse/WFCORE-1984
https://issues.jboss.org/browse/WFCORE-2078
https://issues.jboss.org/browse/WFCORE-2115
https://issues.jboss.org/browse/WFCORE-2239
https://issues.jboss.org/browse/WFCORE-1891
https://issues.jboss.org/browse/WFCORE-2261
https://issues.jboss.org/browse/WFCORE-2262
https://issues.jboss.org/browse/WFCORE-2263

WildFly PR https://github.com/wildfly/wildfly/pull/9595

Experimental Builds: -
Linux https://ci.wildfly.org/viewLog.html?buildId=44066&
Windows https://ci.wildfly.org/viewLog.html?buildId=44068&

Linux experimental build with disabled tests enabled: -
https://ci.wildfly.org/viewLog.html?buildId=44067&
(Expect approximately 91 failures)

### CODE REVIEW

File: remoting/subsystem/src/main/resources/org/jboss/as/remoting/LocalDescriptions.properties [hunk 1]
```diff
@@ -67,18 +67,19 @@ http-connector.security=Configuration of security for this connector.
 http-connector.property=Properties to further configure the connector.
 
 remoting.outbound-connection=Remoting outbound connections.
+remoting.outbound-connection.deprecated=Use the remote-outbound-connection resource instead, with an appropriate protocol.
```
Comment (bstansberry) [inline remoting/subsystem/src/main/resources/org/jboss/as/remoting/LocalDescriptions.properties]: This doesn't do anything by itself. GenericOutboundConnectionResourceDefinition needs to be modified to record the fact it's deprecated, which is what triggers use of this text. Can shift to the constructor variant that uses a Parameters and then the fact of deprecation is set up in the parameters. I'll push something up to illustrate.

This doesn't need to block this PR but if the intent is to deprecate it needs follow up.

File: remoting/subsystem/src/main/resources/org/jboss/as/remoting/LocalDescriptions.properties [hunk 2]
```diff
@@ -91,6 +92,7 @@ remote-outbound-connection.property=The XNIO Options that will be used during th
 
 
 remoting.local-outbound-connection=Remoting outbound connections for local:// URI scheme.
+remoting.local-outbound-connection.deprecated=Use the remote-outbound-connection resource instead, with an appropriate protocol.
```
Comment (bstansberry) [inline remoting/subsystem/src/main/resources/org/jboss/as/remoting/LocalDescriptions.properties]: Same issue as my previous comment.

**Comment by darranl** [comment]
New windows experimental build https://ci.wildfly.org/viewLog.html?buildId=44095&

**Comment by bstansberry** [comment]
#2149 is this plus a commit that addresses the deprecation issue I mentioned above. Feel free to pull that commit into your branch, or if the deprecation wasn't wanted and the text is just cruft ignore it, or this can go in as is and mine can go in later, or @kabir can merge #2149 instead of this one. Kabir, any of these are fine with me, so consider this PR to have my +1.

**Comment by darranl** [comment]
@bstansberry FYI this PR is from a repository within the wildfly-security-incubator organisation so pull requests can be submitted directly against that PR - I will merge yours over there.

**Comment by darranl** [comment]
Latest experimental builds: -
  https://ci.wildfly.org/viewLog.html?buildId=44113&
  https://ci.wildfly.org/viewLog.html?buildId=44114&

**Comment by bstansberry** [comment]
@darranl thanks Darran; I'll close the other one. I sent it up here so if Kabir was dealing with getting this + the full one in while you were away he could just do it without messing with wildfly-security-incubator.  But it's cleaner now, having this all together.

**Comment by darranl** [comment]
A new experimental build: -
  https://ci.wildfly.org/viewLog.html?buildId=44121&
  https://ci.wildfly.org/viewLog.html?buildId=44122&

**Pull request merged**